source/lac/cuda_vector.cu

```diff
@@ -35,51 +35,82 @@ namespace LinearAlgebra
     using ::dealii::CUDAWrappers::block_size;
     using ::dealii::CUDAWrappers::chunk_size;
 
+    namespace
+    {
+      template <typename Number>
+      void
+      delete_device_vector(Number *device_ptr) noexcept
+      {
+        const cudaError_t error_code = cudaFree(device_ptr);
+        (void)error_code;
+        AssertNothrow(error_code == cudaSuccess,
+                      dealii::ExcCudaError(cudaGetErrorString(error_code)));
+      }
+
+      template <typename Number>
+      Number *
+      allocate_device_vector(const std::size_t size)
+      {
+        Number *          device_ptr;
+        const cudaError_t error_code =
+          cudaMalloc(&device_ptr, size * sizeof(Number));
+        (void)error_code;
```

Comment:
The `void` is not necessary. Also you can just write `Utilities::CUDA::malloc(device_ptr, size);` You don't need to change it. I just want you to let you know that this exists :)

Reply:
As long as @bangerth has not approved this PR, I am happy to address things.